pandas/core/internals/construction.py

```diff
@@ -576,6 +576,16 @@ def _convert_object_array(content, columns, coerce_float=False, dtype=None):
     else:
         if len(columns) != len(content):  # pragma: no cover
             # caller's responsibility to check for this...
+            # Its possible that the user may be trying to pass a MultiIndex here.
+            # Attempt to convert columns to MultiIndex, and check length again.
+            # This conversion should be safe to do, as the input expects an Index
+            # and the colums are left unmodified throughout the rest of this function.
+            columns = ensure_index(columns)
+            if len(columns) != content_length:
```

Comment:
Could you just combine this check with the one above so you can reuse the same error message?